### PR TITLE
PR 2280 followups

### PR DESCRIPTION
**Pull request by myronmarston**
@eugeneius -- can you review my refactoring here and confirm my refactoring here is functionality equivalent to what you had before?

### CODE REVIEW

File: lib/rspec/core/configuration.rb [hunk 1]
```diff
@@ -2031,21 +2026,32 @@ def configure_group_with(group, module_list, application_method)
         end
       end
 
-      def on_existing_matching_groups(meta, scope=:ignore)
-        groups = world.example_groups.dup
-
-        until groups.empty?
-          group = groups.shift
+      def add_hook_to_exisiting_matching_groups(meta, scope, &block)
```
Comment (eugeneius) [inline lib/rspec/core/configuration.rb]: exisiting -> existing

**Comment by eugeneius** [comment]
This looks great - traversing the example group hierarchy is much clearer 👍

File: lib/rspec/core/configuration.rb (hunk 1 above)
Comment (myronmarston) [inline lib/rspec/core/configuration.rb]: Good catch, thanks.

**Comment by myronmarston** [comment]
@rspec/rspec can someone review this?  Then I'll merge and get this (and #2280) released in 3.5.1.

**Comment by JonRowe** [comment]
Good work!

**Pull request merged**